server/src/main/java/org/elasticsearch/cluster/metadata/ReservedStateMetadata.java

```diff
@@ -283,6 +283,13 @@ public Builder putHandler(ReservedStateHandlerMetadata handler) {
             return this;
         }
 
+        /**
+         * Returns the current handler metadata stored in the builder
+         */
+        public ReservedStateHandlerMetadata getHandler(String handlerName) {
```

Comment:
Nit: maybe this is better named `getHandlerMetadata`?

Reply:
I'll address this in a followup, it will increase the size of this PR because the method is used in many tests.